frontend/svelte/src/tests/lib/utils/accounts.utils.spec.ts

```diff
@@ -26,7 +27,8 @@ class MockLedgerCanister extends LedgerCanister {
 describe("accounts-utils", () => {
   beforeAll(() => {
     // Needed to prevent importing Http from @dfinity/agent
-    const mockCreateAgent = () => undefined;
+    const mockCreateAgent = () =>
+      new Promise<HttpAgent>((resolve) => resolve(undefined));
```

Comment:
```suggestion
    const mockCreateAgent = Promise.resolve(undefined)
```

Is this actually being used? I'm surprised not to see any changes at call sites

Reply:
We mock the `LedgerCanister.create`, therefore the `createAgent` is never used in the tests.

```
jest
      .spyOn(LedgerCanister, "create")
      .mockImplementation((): LedgerCanister => mockLedgerCanister);
```

Yet, to avoid a dependency on fetch when creating the agent from `@dfinity/agent`, we need to avoid importing it. Without a mockImplementation, it imports the file, also the library and the error appears.

```
    Fetch implementation was not available. You appear to be in a Node.js context, but global.fetch was not available.

      4 | // To avoid being executed in tests that only import it
      5 | export const createAgent = async () => {
    > 6 |   const agent = new HttpAgent({ host: serviceURL });
        |                 ^
      7 |
      8 |   if (process.env.FETCH_ROOT_KEY === "true") {
      9 |     await agent.fetchRootKey();

      at getDefaultFetch (node_modules/@dfinity/agent/src/agent/http/index.ts:100:13)
      at new HttpAgent (node_modules/@dfinity/agent/src/agent/http/index.ts:147:38)
      at createAgent (src/lib/utils/agent.utils.ts:6:17)
      at loadAccounts (src/lib/utils/accounts.utils.ts:12:29)
      at Object.<anonymous> (src/tests/lib/utils/accounts.utils.spec.ts:41:23)
```

Reply:
I'm still a bit confused about why having `mockCreateAgent` makes an error about `global.fetch` not existing disappear, but I don't have much experience with typescript :)